core/sql/sqlcomp/PrivMgrMD.cpp

```diff
@@ -962,29 +979,41 @@ PrivStatus PrivMgrMDAdmin::getReferencingTablesForConstraints (
     trafMetadataLocation_ + ".OBJECTS o";
   std::string tblConstraintsMDTable = 
     trafMetadataLocation_ + ".TABLE_CONSTRAINTS t";
+  std::string keysMDTable = 
+    trafMetadataLocation_ + ".KEYS k";
   std::string uniqueRefConstraintsMDTable = 
     trafMetadataLocation_ + ".UNIQUE_REF_CONSTR_USAGE u";
 
   // Select all the constraints that are referenced by the table
-  // create_time is included to order by the oldest to newest
   std::string selectStmt = "select distinct o.object_uid, o.object_owner, o.object_type, o.create_time, ";
   selectStmt += "trim(o.catalog_name) || '.\"' || ";
-  selectStmt += "trim (o.schema_name) || '\".\"' ||";
-  selectStmt += "trim (o.object_name)|| '\"' from ";
+  selectStmt +=   "trim (o.schema_name) || '\".\"' ||";
+  selectStmt +=   "trim (o.object_name)|| '\"' ";
+  selectStmt += ", c.column_number "; 
+  selectStmt += "from " + uniqueRefConstraintsMDTable + std::string(", ");
```

Comment:
Why is that ","  should be std::string?  I wonder if std::string will trigger memory allocation in the malloc heap space